libs/npm_installer/bin_entries.rs

```diff
@@ -487,6 +495,16 @@ fn symlink_bin_entry<'a>(
     pathdiff::diff_paths(to, from)
   }
 
+  let original_relative = relative_path(bin_node_modules_dir_path, &original)
+    .map(Cow::Owned)
+    .unwrap_or_else(|| Cow::Borrowed(&original));
+
+  if let Ok(original_link) = sys.fs_read_link(&link)
+    && *original_link == *original_relative
+  {
+    return Ok(EntrySetupOutcome::Success);
+  }
```

Comment:
Maybe this should only happen on error? I've found reading symlinks to be really slow compared to just setting up the symlinks.

Reply:
Really? On my machine a successful `symlink` is about 50x slower than a readlink. And a failed symlink because the link is already there is about 15x slower than a readlink

Reply:
Interesting. IIRC reading is slower than writing on Windows as well. I guess it doesn't matter too much because there won't be a lot of these, as you mentioned.